enhancements/metering/user-provided-pricesheets.md

```diff
@@ -0,0 +1,205 @@
+
+We start by enabling users to define price sheets by documenting how to do so with existing functionality that's available, and by defining a few more default ReportQueries that utilize price sheets when calculating costs.
+
+At a high level this means we need to do the following:
+
+- Define a Presto table schema that supports both user-provided prices and calculated prices from cloud billing data.
+- Create built in [ReportQuery][reportqueries] and [ReportDataSource][reportdatasources] that can create price sheet table using AWS billing data.
+- Create built in [ReportQueries][reportqueries] that calculate costs using a price sheet specified as a [Report input][report-input].
+
+### User Stories [optional]
+
+- As a member of the operations team, I want to create a Report that let's me calculate costs with any price sheet.
+- As a member of the operations team, I want to be able to define my own prices to be used with Reports.
+
+### Implementation Details/Notes/Constraints [optional]
+
+
+#### Price Sheet Table Schema
+
+The Presto table schema is defining a set of column names and their data types that all price sheet Presto database tables will have, enabling ReportQueries designed to use price sheets to work with any of them interchangeably.
+
+The price sheet will define 7 columns.
+
+- `resource varchar` is the type of resource, such as CPU or Memory
+- `resource_unit varchar` is the unit of the resource. For CPU it might be cores or millicores and for memory it could be bytes, gigabytes, etc.
+- `time_unit varchar` is the unit of time that the resource is billed at. For example 'hour' or 'minute'.
+- `price double` is the price the resource is billed at, in terms of resources used in terms of `resource_unit` per `time_unit`.
+- `currency varchar` is the currency the price is in.
+- `price_start_date timestamp` is when this price starts
+- `price_end_date timestamp` is when this price ends
+
+There are a few properties that each row must obey to ensure correct results:
+- Each row is expected to be a non-overlapping period of time specified by `price_start_dateand` and `price_end_date`, also known as the "price period".
+- For each price period, there is only 1 row for a particular value of `resource`. This means one currency, one price, one `resource_unit`, one time_unit` per price period per resource.
+- For a given `resource`, the correct `unit` is specified. Eg: for `resource=cpu`, `resource_unit` must be either `cores` or `millicores`, and must not be specified in different units within the same price period.
+
+
+#### Initial design
+
+To start, we will provide an example of a ReportDataSource and ReportQuery users can use to create a price sheet (details in ReportDataSources) and a set of ReportQueries for calculating cost using a price sheet ReportDataSource.
+Once we are satisfied with the results we can enable creating these resource as part of the Metering install by configuring any necessary options in the MeteringConfig.
```

Comment:
"Once we are satisfied with the results" -> "Once we are satisfied with the results,"